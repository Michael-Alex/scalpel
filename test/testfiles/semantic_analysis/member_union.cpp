class c
{
	union inner
	{
		double d;
		int i;
	};
};

union u
{
	union inner
	{
		double d;
		int i;
	};

	char c;
	inner i;
};

u::inner ui;

