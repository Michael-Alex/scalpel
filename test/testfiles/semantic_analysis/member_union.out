{
  "id": 0,
  "classes":
  [
    {
      "id": 0,
      "name": "c",
      "unions":
      [
        {
          "id": 0,
          "name": "inner",
          "access": "private",
          "destructor":
          {
            "access": "public"
          },
          "variables":
          [
            {
              "id": 0,
              "name": "d",
              "access": "public",
              "type":
              {
                "fundamental type": "double"
              }
            },
            {
              "id": 1,
              "name": "i",
              "access": "public",
              "type":
              {
                "fundamental type": "int"
              }
            }
          ]
        }
      ],
      "destructor":
      {
        "access": "public"
      }
    }
  ],
  "unions":
  [
    {
      "id": 0,
      "name": "u",
      "unions":
      [
        {
          "id": 1,
          "name": "inner",
          "access": "public",
          "destructor":
          {
            "access": "public"
          },
          "variables":
          [
            {
              "id": 2,
              "name": "d",
              "access": "public",
              "type":
              {
                "fundamental type": "double"
              }
            },
            {
              "id": 3,
              "name": "i",
              "access": "public",
              "type":
              {
                "fundamental type": "int"
              }
            }
          ]
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "variables":
      [
        {
          "id": 4,
          "name": "c",
          "access": "public",
          "type":
          {
            "fundamental type": "char"
          }
        },
        {
          "id": 5,
          "name": "i",
          "access": "public",
          "type":
          {
            "member union": 1
          }
        }
      ]
    }
  ],
  "variables":
  [
    {
      "id": 0,
      "name": "ui",
      "type":
      {
        "member union": 1
      }
    }
  ]
}
